### ISSUE TITLE
Parser fails to parse schema qualified type in CREATE TYPE .. AS ENUM commands

### ISSUE THREAD

**Issue by jonathan-meier** [jOOQ/jOOQ#10852]
### Expected behavior

Parser successfully parses command
```
CREATE TYPE my_schema.my_enum AS ENUM ('MY_ENUM_1', 'MY_ENUM_2');
```
### Actual behavior

A `ParserException` is thrown:
```
org.jooq.impl.ParserException: Keyword 'AS ENUM' expected: [1:22] CREATE TYPE my_schema[*].my_enum AS ENUM ('MY_ENUM_1', 'MY_ENUM_2');
        at org.jooq.impl.ParserContext.expected(ParserImpl.java:12151)
        at org.jooq.impl.ParserImpl.parseKeyword(ParserImpl.java:11600)
        at org.jooq.impl.ParserImpl.parseCreateType(ParserImpl.java:4168)
        at org.jooq.impl.ParserImpl.parseCreate(ParserImpl.java:2485)
        at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:914)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:614)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:601)
        at org.jooq.meta.extensions.ddl.DDLDatabase.load(DDLDatabase.java:175)
        at org.jooq.meta.extensions.ddl.DDLDatabase.access$000(DDLDatabase.java:86)
        at org.jooq.meta.extensions.ddl.DDLDatabase$2.load(DDLDatabase.java:160)
        at org.jooq.FilePattern.load0(FilePattern.java:306)
        at org.jooq.FilePattern.load(FilePattern.java:286)
        at org.jooq.FilePattern.load(FilePattern.java:232)
        at org.jooq.meta.extensions.ddl.DDLDatabase.export(DDLDatabase.java:157)
```

### Steps to reproduce the problem

Use the code generator with the `DDLDatabase`:
```
generator {
	database {
		name 'org.jooq.meta.extensions.ddl.DDLDatabase'

		properties {
			property {
				key 'scripts'
				value 'my_script.sql'
			}
		}
	}
}
```
where `my_script.sql` contains the `CREATE TYPE` command above.

### Version

- jOOQ: 3.14

### Additional info

Indeed, when parsing `CREATE TYPE` commands only an identifier is parsed for the type name instead of a name:
https://github.com/jOOQ/jOOQ/blob/286635b8eacaf71bd42abad051471b9a76b487e7/jOOQ/src/main/java/org/jooq/impl/ParserImpl.java#L4166-L4181

Is this just an oversight (bug), or is there a reason for not parsing a (schema qualified) name here?

**Comment by lukaseder** [comment]
Thanks a lot for your report. I can reproduce this. Will fix right away.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.15.0 and 3.14.2 (#10855)
